plugins/woocommerce/src/Database/Migrations/MigrationHelper.php

```diff
@@ -130,110 +136,116 @@ private static function migrate_country_states_for_shipping_locations( string $c
+				$update_query = $wpdb->prepare(
+					"UPDATE {$wpdb->prefix}wc_order_addresses SET state=%s WHERE country=%s AND state=%s LIMIT %d",
+					$new_state,
+					$country_code,
+					$old_state,
+					$limit
+				);
+
+				$wpdb->query( $update_query );
+			}
 
-		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-		$sql = $wpdb->prepare(
-			"SELECT meta_id, post_id, meta_value FROM {$wpdb->prefix}postmeta
+			$update_query = $wpdb->prepare(
+				"UPDATE {$wpdb->prefix}postmeta
+				SET meta_value=%s
 				WHERE (meta_key='_billing_state' OR meta_key='_shipping_state')
-				AND meta_value IN {$states_as_comma_separated}
+				AND meta_value=%s
 				AND post_id IN (
 					SELECT post_id FROM {$wpdb->prefix}postmeta WHERE
 					(meta_key = '_billing_country' OR meta_key='_shipping_country')
 					AND meta_value=%s
-				)",
-			$country_code
-		);
-		// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-
-		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
-		$addresses_data = $wpdb->get_results( $sql, ARRAY_A );
-
-		foreach ( $addresses_data as $address_data ) {
-			$update_query = $wpdb->prepare(
-				"UPDATE {$wpdb->prefix}postmeta SET meta_value=%s WHERE meta_id=%d",
-				$old_to_new_states_mapping[ $address_data['meta_value'] ],
-				$address_data['meta_id']
+				)
+				LIMIT %d",
+				$new_state,
+				$old_state,
+				$country_code,
+				$limit
 			);
-			$wpdb->query( $update_query );
 
-			$update_query = $wpdb->prepare(
-				"UPDATE {$wpdb->prefix}posts SET post_modified=%s, post_modified_gmt=%s WHERE ID=%d",
-				$current_date,
-				$current_date_gmt,
-				$address_data['post_id']
-			);
 			$wpdb->query( $update_query );
 		}
-		//phpcs:enable WordPress.DB.PreparedSQL.NotPrepared
-	}
 
-	/**
-	 * Migrate the state code for the store location.
-	 *
-	 * @param string $country_code The country that has the states for which the migration is needed.
-	 * @param array  $old_to_new_states_mapping An associative array where keys are the old state codes and values are the new state codes.
-	 * @return void
-	 */
-	private static function migrate_country_states_for_store_location( string $country_code, array $old_to_new_states_mapping ): void {
-		$store_location = get_option( 'woocommerce_default_country', '' );
-		if ( StringUtil::starts_with( $store_location, "{$country_code}:" ) ) {
-			$old_location_code = substr( $store_location, 3 );
-			if ( array_key_exists( $old_location_code, $old_to_new_states_mapping ) ) {
-				$new_location_code = "{$country_code}:{$old_to_new_states_mapping[$old_location_code]}";
-				update_option( 'woocommerce_default_country', $new_location_code );
-			}
+		$states_as_comma_separated = "('" . join( "','", array_keys( $old_to_new_states_mapping ) ) . "')";
+
+		$posts_exist_query = $wpdb->prepare(
+			"
+			SELECT 1 FROM {$wpdb->prefix}postmeta
+			WHERE (meta_key='_billing_state' OR meta_key='_shipping_state')
+			AND meta_value IN {$states_as_comma_separated}
```

Comment:
Just curious, is there a reason why this var is interpolated while the `$country_code` is `prepare()`-d? Is it because country code can contain `%`?

Reply:
The country code is a simple string, it works fine with `prepare` and `%s`. `$states_as_comma_separated`, however, is a comma-separated and parenthesized list of strings, e.g. `('NL', 'WA', 'AK',...)` and needs to be interpolated; it's the same case as `{$wpdb->prefix}`.